linux/LinuxMachine.c

```diff
@@ -603,6 +603,48 @@ static void scanCPUFrequencyFromCPUinfo(LinuxMachine* this) {
    }
 }
 
+#ifdef HAVE_SENSORS_SENSORS_H
+static void LinuxMachine_fetchCPUTopologyFromCPUinfo(LinuxMachine* this) {
+   const Machine* super = &this->super;
+
+   FILE* file = fopen(PROCCPUINFOFILE, "r");
+   if (file == NULL)
+      return;
+
+   int cpuid = -1;
+   int coreid = -1;
+   int physicalid = -1;
+
+   while (!feof(file)) {
+      char buffer[PROC_LINE_LENGTH];
+
+      if (fgets(buffer, PROC_LINE_LENGTH, file) == NULL)
+         break;
+
+      if (sscanf(buffer, "processor : %d", &cpuid) == 1) {
+         continue;
+      } else if (sscanf(buffer, "physical id : %d", &physicalid) == 1) {
+         continue;
+      } else if (sscanf(buffer, "core id : %d", &coreid) == 1) {
+         continue;
```

Comment:
Consider using prefix check (`String_startsWith` from `XUtils.h`) with `fast_strtoull`.

Reply:
I want to preserve the whitespace skipping of sscanf for robustness, gating it with a prefix check now.